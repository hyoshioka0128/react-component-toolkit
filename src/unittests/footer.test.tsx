// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Footer from "../components/footer/footer";
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, screen, waitFor } from '@testing-library/react'
import {faker} from '@faker-js/faker'


const componentId:string = "ComponentUnderTest";
let fakePrelinkText: string = '';
let fakeLink: string = '';
let fakeLinkText:string = '';
let testLeftFooterText:string = '';

describe("Testing component : <Footer />", () => {
  beforeAll(() => {
  });

  afterAll(() => {
  })
  
  beforeEach(() => {
    fakePrelinkText= faker.lorem.words(2);
    fakeLink = faker.internet.url();
    fakeLinkText= faker.lorem.words(2);
    testLeftFooterText= faker.lorem.words(2);
  });

  afterEach(() => {
    fakePrelinkText = '';
    fakeLink = '';
    fakeLinkText = '';
    testLeftFooterText = '';
  });

  

  test("<Footer /> should render with id", async () => {

    // Create our component with our generated id
    const testRenderer = renderer.create(<Footer id={componentId} footerTextLeft='' footerPreLink='' footerLink='' footerLinkText=''/>);

    // Use the created renderer to convert to json and then check it matches our expected snapshot
    const jsonSnapshot = testRenderer.toJSON();

    // Check that our rendered component does have the exact randomly generated componentid for this test
    expect(jsonSnapshot).toHaveProperty("props.id", componentId);

    // Match that our rendered component matches our snapshot of the component and ignore the random id
    expect(jsonSnapshot).toMatchSnapshot(
      {
        props: { id: expect.any(String) }
      }
    );

    expect(jsonSnapshot).toMatchSnapshot();

  });

  test("<Footer /> test for styling", async () => {
    // Create our component with our generated id
    const testRenderer = renderer.create(<Footer id={componentId} footerTextLeft='' footerPreLink='' footerLink='' footerLinkText='' />);

    // Use the created renderer to convert to json and then check it matches our expected snapshot
    const jsonSnapshot = testRenderer.toJSON();

    expect(jsonSnapshot).toHaveStyleRule("font-family", "SofiaPro");
  
    expect(jsonSnapshot).toHaveStyleRule("font-size", "14px");

    expect(jsonSnapshot).toHaveStyleRule("font-weight", "500");
  
    expect(jsonSnapshot).toHaveStyleRule("line-height", "1.2");

    expect(jsonSnapshot).toHaveStyleRule("letter-spacing", "normal");

  });

  test("<Footer /> test left footer text", async () => {
    render(<Footer id={componentId} footerTextLeft={testLeftFooterText} footerPreLink={fakePrelinkText} footerLink={fakeLink} footerLinkText={fakeLinkText}/>);  
    expect(screen.getByText(testLeftFooterText).textContent).toMatch(testLeftFooterText);
  });


  test("<Footer /> test right footer text", async () => {
    render(<Footer id={componentId} footerTextLeft={testLeftFooterText} footerPreLink={fakePrelinkText} footerLink={fakeLink} footerLinkText={fakeLinkText} />);
    expect(screen.getByText(fakePrelinkText).textContent).toMatch(fakePrelinkText);
    expect(screen.getByText(fakeLinkText).textContent).toMatch(fakeLinkText);
    expect(screen.getByRole('link')).toHaveAttribute('href', fakeLink);
    
  })

  

});