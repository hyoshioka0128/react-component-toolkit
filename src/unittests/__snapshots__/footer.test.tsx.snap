// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component : <Footer /> <Footer /> should render with id 1`] = `
{
  "children": [
    .c0.c0.c0 {
  color: #93268f !important;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
}

<div
      className="mantine-Grid-root mantine-nsdj"
    >
      <div
        className="mantine-Grid-col mantine-MediaQuery-media mantine-kre0hd"
      />
      <div
        className="mantine-Grid-col mantine-MediaQuery-media mantine-7v2dpu"
      >
         
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways c0 css-1bhi0za-MuiTypography-root-MuiLink-root"
          href=""
          onBlur={[Function]}
          onFocus={[Function]}
        />
      </div>
    </div>,
  ],
  "props": {
    "className": "sc-beqWaB eAPmUo",
    "data-testid": "FooterTestId",
    "id": Any<String>,
  },
  "type": "div",
}
`;

exports[`Testing component : <Footer /> <Footer /> should render with id 2`] = `
.c0 {
  id: ComponentUnderTest;
  font-family: SofiaPro;
  font-size: 14px;
  font-weight: 500;
  line-height: 1.2;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c1.c1.c1 {
  color: #93268f !important;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
}

<div
  className="c0"
  data-testid="FooterTestId"
  id="ComponentUnderTest"
>
  <div
    className="mantine-Grid-root mantine-nsdj"
  >
    <div
      className="mantine-Grid-col mantine-MediaQuery-media mantine-kre0hd"
    />
    <div
      className="mantine-Grid-col mantine-MediaQuery-media mantine-7v2dpu"
    >
       
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways c1 css-1bhi0za-MuiTypography-root-MuiLink-root"
        href=""
        onBlur={[Function]}
        onFocus={[Function]}
      />
    </div>
  </div>
</div>
`;
